app/screens/create_direct_message/create_direct_message.tsx

```diff
@@ -87,307 +42,70 @@ export default function CreateDirectMessage({
     currentUserId,
     restrictDirectMessage,
     teammateNameDisplay,
-    tutorialWatched,
 }: Props) {
     const serverUrl = useServerUrl();
-    const theme = useTheme();
-    const style = getStyleFromTheme(theme);
     const intl = useIntl();
-    const {formatMessage} = intl;
-
-    const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
-    const next = useRef(true);
-    const page = useRef(-1);
-    const mounted = useRef(false);
 
-    const [profiles, dispatchProfiles] = useReducer(reduceProfiles, []);
-    const [searchResults, setSearchResults] = useState<UserProfile[]>([]);
-    const [loading, setLoading] = useState(false);
-    const [term, setTerm] = useState('');
-    const [startingConversation, setStartingConversation] = useState(false);
     const [selectedIds, setSelectedIds] = useState<{[id: string]: UserProfile}>({});
-    const selectedCount = Object.keys(selectedIds).length;
-
-    const isSearch = Boolean(term);
-
-    const loadedProfiles = ({users}: {users?: UserProfile[]}) => {
-        if (mounted.current) {
-            if (users && !users.length) {
-                next.current = false;
-            }
-
-            page.current += 1;
-            setLoading(false);
-            dispatchProfiles({type: 'add', values: users});
-        }
-    };
-
-    const clearSearch = useCallback(() => {
-        setTerm('');
-        setSearchResults([]);
-    }, []);
-
-    const getProfiles = useCallback(debounce(() => {
-        if (next.current && !loading && !term && mounted.current) {
-            setLoading(true);
-            if (restrictDirectMessage) {
-                fetchProfilesInTeam(serverUrl, currentTeamId, page.current + 1, General.PROFILE_CHUNK_SIZE).then(loadedProfiles);
-            } else {
-                fetchProfiles(serverUrl, page.current + 1, General.PROFILE_CHUNK_SIZE).then(loadedProfiles);
-            }
-        }
-    }, 100), [loading, isSearch, restrictDirectMessage, serverUrl, currentTeamId]);
-
-    const handleRemoveProfile = useCallback((id: string) => {
-        const newSelectedIds = Object.assign({}, selectedIds);
-
-        Reflect.deleteProperty(newSelectedIds, id);
-
-        setSelectedIds(newSelectedIds);
-    }, [selectedIds]);
+    const [page, setPage] = useState(-1);
```

Comment:
Do you need a re-render and that kind of persistence for the page? Or can this be again a reference? If it is a reference, the child can still update the value if needed (0/5 on who should be the one responsible, though).

Reply:
I could net get the reference to work :( I'll take another look!

Reply:
I must be missing something simple here.  I have tried the following to pass the ref from the parent to the child.

In `CreatDirectMessage`: 
*  const pageRef = useRef(-1);
* pass as prop to `UsersModal`

In `UsersModal`
* use `forwardRef` on the compent
* <img width="554" alt="image" src="https://user-images.githubusercontent.com/7575921/201413223-8201b0b7-7d42-4145-ae82-674f948ee555.png">

* change the value by accessing `pageRef.current +=1`

Reply:
Even if the name is ref, there is no need to forward them as ref the way you are doing it. The `forwardRef` pattern is in order to modify the default behavior of the `ref` prop, in general so the ref does not reference this component but a component lower in the tree (e.g. when you do a wrapper of an input, and want the ref to behave as an input).

Here I am just saying to pass it as a normal prop. For example...
```
const ParentComponent = () => {
    const page = useRef(-1);
    return <ChildComponent page={page} />
}

type ChildProps = {
    page: React.MutableRefObject<number>,
}

const ChildComponent = ({page}: ChildProps) => {
    page.current += 1;
    return (...)
}
```